### ISSUE TITLE
How to integrate D3 into this generator's build?

### ISSUE THREAD

**Issue by realisation** [mcfly-io/generator-mcfly#308]
I'm working on a legacy app and I need to install d3. I tried to make it work via dependency injection like this: http://www.ng-newsletter.com/posts/d3-on-angular.html but had no luck. I also tried just placing the script tag on the index html with the path it would have when compiled. No luck there either.

Would appreciate any guidance on this - this is for a pro bono civic hacking app.

**Comment by thaiat** [comment]
what error do you have with this approach:
```js
angular.module('d3', [])
  .factory('d3Service', ['$document', '$q', '$rootScope',
    function($document, $q, $rootScope) {
      var d = $q.defer();
      function onScriptLoad() {
        // Load client in the browser
        $rootScope.$apply(function() { d.resolve(window.d3); });
      }
      // Create a script tag with d3 as the source
      // and call our onScriptLoad callback when it
      // has been loaded
      var scriptTag = $document[0].createElement('script');
      scriptTag.type = 'text/javascript'; 
      scriptTag.async = true;
      scriptTag.src = 'http://d3js.org/d3.v3.min.js';
      scriptTag.onreadystatechange = function () {
        if (this.readyState == 'complete') onScriptLoad();
      }
      scriptTag.onload = onScriptLoad;
 
      var s = $document[0].getElementsByTagName('body')[0];
      s.appendChild(scriptTag);
 
      return {
        d3: function() { return d.promise; }
      };
}]);
```

**Comment by realisation** [comment]
No error, just no behavior. I did all the work with making a d3 module to inject into the main app to then use the service from the d3 module, and when I put the directive tag in HTML there was just no behavior. 

Right this moment I'm trying to have the service merely return a string to see if the service is working at all.

**Comment by realisation** [comment]
So it is at least returning a string....

**Comment by thaiat** [comment]
kind of difficult to help you ... can you try in chrome debugger and see if you have access to the global d3 object

**Comment by realisation** [comment]
Hm strange, it is giving me an error `GET http://localhost:5000/node_modules/d3/d3.js`. I must have overlooked this earlier when I was checking to see if it was in node_modules folder in the sources, which it was, so this is a little confusing it can't find it because it's there.

**Comment by thaiat** [comment]
the app in the browser is served from the dist folder, so it does not have access to node_modules
What you need to do is in the begining of `main.js` file:
```js
var d3 = require('d3');
``` 
or if you want it globally
```js
window.d3 = require('d3');
```

**Comment by realisation** [comment]
Yeah, I mean it is in the node modules according to the chrome dev tool sources tab. So it's there :/ - I have `require('d3')` in the index of both the modules, and a script tag with the path to node_module from the compiled version. 

I just tried putting it in main.js as just d3 and window.d3, but they didn't work, at least with the sample directive I tried to implement from that angular newsletter post.

It is present on the window object, however.
